File: src/ros2_bridge/mappers/foxglove_msgs/geo_json.rs
//! Typed mapper for `foxglove_msgs/msg/GeoJSON`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn geo_json_to_bus(msg: ros_env::foxglove_msgs::msg::GeoJSON) -> crate::foxglove_msgs::msg::v1::GeoJson {
    crate::foxglove_msgs::msg::v1::GeoJson {
        geojson: crate::ros2_bridge::mappers::convert::from_ros_string(msg.geojson),
    }
}

pub(crate) fn geo_json_to_ros(bus: crate::foxglove_msgs::msg::v1::GeoJson) -> ros_env::foxglove_msgs::msg::GeoJSON {
    ros_env::foxglove_msgs::msg::GeoJSON {
        geojson: crate::ros2_bridge::mappers::convert::to_ros_string(bus.geojson),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct FoxgloveMsgsGeoJsonMapper;

impl TypedTopicMapper for FoxgloveMsgsGeoJsonMapper {
    type Ros = ros_env::foxglove_msgs::msg::GeoJSON;
    type Bus = crate::foxglove_msgs::msg::v1::GeoJson;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(geo_json_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(geo_json_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/foxglove_msgs/geo_json.py
"""Generated mapper for `foxglove_msgs/msg/GeoJSON`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.foxglove_msgs.msg.v1 import GeoJSON as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def geo_json_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.geojson = str(msg.geojson)
    return bus


def geo_json_to_ros(bus):
    from foxglove_msgs.msg import GeoJSON as RosMsg

    out = RosMsg()
    out.geojson = str(bus.geojson)
    return out


class FoxgloveMsgsGeoJsonMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from foxglove_msgs.msg import GeoJSON as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return geo_json_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return geo_json_to_ros(bus)
